### PR TITLE
Enable sqlclosecheck linter

### PR DESCRIPTION
**Pull request by pellared**
## Why

Per https://github.com/ryanrolds/sqlclosecheck/releases/tag/v0.4.0

## What

- Enable https://github.com/ryanrolds/sqlclosecheck
- Address the reported issues (close `*sql.Rows` and `*sql.Stmt`)
- Use `assert` instead of `require` in `t.Cleanup` to make sure that that everything is cleaned up (`require` calls `t.Fatal/FailNow` which calls `runtime.Goexit`). As far as I remember the problem would be only if multiple `require` are called in a single callback passed to `t.Cleanup`, but I prefer to use `assert` consistently to not burden my mind each time I see `require` together with `t.Cleanup`. Example [here](https://github.com/signalfx/splunk-otel-go/pull/1754/files#r1098531496)

### CODE REVIEW

**Comment by pellared** [comment]
@MrAlias I think that the lint failures in `integration_test.go` files makes sense. In `suite_test.go` I think we could add `//nolint:` for the whole package. WDYT? If you agree I can work on address the linting issues.

**Comment by MrAlias** [comment]
> @MrAlias I think that the lint failures in `integration_test.go` files make sense. In `suite_test.go` I think we could add `//nolint:` for the whole package. WDYT? If you agree I can work on address the linting issues.

Why not just fix the errors by closing the appropriate values?

**Comment by pellared** [comment]
> Why not just fix the errors by closing the appropriate values?

Works for me.

**Comment by pellared** [comment]
@MrAlias PTAL

File: instrumentation/github.com/go-sql-driver/mysql/splunkmysql/test/integration_test.go [hunk 1]
```diff
@@ -64,8 +64,8 @@ func newFixtures(t *testing.T) (*tracetest.SpanRecorder, *trace.TracerProvider,
 	db, err := splunksql.Open("mysql", dsn, splunksql.WithTracerProvider(tp))
 	require.NoError(t, err)
 	t.Cleanup(func() {
-		require.NoError(t, db.Close())
-		require.NoError(t, tp.Shutdown(context.Background()))
```
Comment (pellared) [inline instrumentation/github.com/go-sql-driver/mysql/splunkmysql/test/integration_test.go:68]: This one would not be called if the assertion in previous line fails.

**Review by MrAlias — APPROVED** [review]

**Pull request merged**